modules/assim.batch/R/pda.emulator.R

```diff
@@ -86,76 +83,205 @@ pda.emulator <- function(settings, params.id=NULL, param.names=NULL, prior.id=NU
   knots.probs <- knots.list$probs
   
   
-  ## Set up runs and write run configs for all proposed knots X
-  run.ids <- pda.init.run(settings, con, my.write.config, workflow.id, knots.params, 
-                          n=settings$assim.batch$n.knot, 
-                          run.names=paste0("Knot.",1:settings$assim.batch$n.knot))
-
-  ## start model runs
-  start.model.runs(settings,settings$database$bety$write)
-
-  ## Retrieve model outputs, calculate likelihoods (and store them in database)
-  LL.X <- rep(NA, settings$assim.batch$n.knot)
-  model.out <- list()
+  ## Check which emulator extension type requested if any
+  if(!is.null(settings$assim.batch$extension)){
+    
+    if(settings$assim.batch$extension == "round"){
+      
+      # save the original prior path
+      temp.path = settings$assim.batch$prior$path
+      
+      # set prior path to NULL to use the previous PDA's posterior densities as new priors this time
+      settings$assim.batch$prior$path = NULL
+      
+      ## Re-load priors
+      temp <- pda.load.priors(settings, con) # loads the posterior dist. from previous emulator run
+      prior <- temp$prior
+      settings$assim.batch$prior$path = temp.path
+      
+      ## Re-set prior distribution functions 
+      prior.fn <- pda.define.prior.fn(prior)
+      
+      ## Propose 75% of the new parameter knots from the posterior of previous run
+      knot.par <- ifelse(!is.null(settings$assim.batch$knot.par),
+                         as.numeric(settings$assim.batch$knot.par),
+                         0.75)
```

Comment:
Ah, OK, I see that now. It might still be a good idea to define it as a named global constant somewhere at the beginning of the function though, although it might not matter as much here because it's well documented in the comment.